### PR TITLE
kubectl describe pod: use ReportingController as an event source

### PR DESCRIPTION
**Pull request by ingvagabund**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
With https://github.com/kubernetes/kubernetes/pull/91645 merged (promotion of v1beta1 events to v1),
it's no longer permitted to set Source field.
Based on https://stupefied-goodall-e282f7.netlify.app/contributors/design-proposals/instrumentation/events-redesign/#backward-compatibility
Source.Component is supposed to be replaced by ReportingController.
Source.Host is supposed to be replaced by DeprecatedHost, though DeprecatedHost
was never implemented. Thus, ignoring the second field.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
Following changes in https://github.com/kubernetes/kubernetes/pull/92874/files#diff-756d0996ce76e743d8812e2afecf746dR107-R110

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/describe/describe.go [hunk 1]
```diff
@@ -5001,15 +5005,6 @@ func translateTimestampSince(timestamp metav1.Time) string {
 	return duration.HumanDuration(time.Since(timestamp.Time))
 }
 
-// formatEventSource formats EventSource as a comma separated string excluding Host when empty
-func formatEventSource(es corev1.EventSource) string {
-	EventSourceString := []string{es.Component}
-	if len(es.Host) > 0 {
```
Comment (zhouya0) [inline staging/src/k8s.io/kubectl/pkg/describe/describe.go:5007]: Omit `EventSource.Host` in such way, should we care backward compatibility?

Comment (ingvagabund) [reply staging/src/k8s.io/kubectl/pkg/describe/describe.go:5007]: Host is deprecated in 1.18 or less, can't be set in 1.19. How far back in the past kubectl compatibility goes? 2 or 3 versions back?

**Comment by ingvagabund** [comment]
@zhouya0 @brianpursley PTAL

File: staging/src/k8s.io/kubectl/pkg/describe/describe.go (hunk 1 above)
Comment (soltysh) [reply staging/src/k8s.io/kubectl/pkg/describe/describe.go:5007]: Given this is targeting 1.20, and you can't set that in 1.19 this ok change.

Comment (soltysh) [reply staging/src/k8s.io/kubectl/pkg/describe/describe.go:5007]: We support only +/- 1 version

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
/milestone v1.20-phase-bug

**Pull request merged**